backend/portal/views.py

```diff
@@ -277,3 +277,37 @@ def review(self, request):
                 many=True,
             ).data
         )
+
+    @action(detail=False, methods=["get"])
+    def status(self, request):
+        # awaiting approval: no admin comment, not approved
+        posts_awaiting_approval = PostSerializer(
+            Post.objects.filter(
+                (Q(admin_comment=None) | Q(admin_comment="")),
+                Q(user=request.user),
```

Comment:
for the `user=request.user`, i don't think you need to wrap the Q object around it. we only use Q objects for advanced searches (or's, __in's, etc.)